src/ripple/rpc/impl/ServerHandlerImp.h

```diff
@@ -132,15 +132,12 @@ class ServerHandlerImp
         boost::asio::ip::tcp::endpoint endpoint);
 
     Handoff
-    onHandoff (Session& session,
+    onHandoff(
+        Session& session,
         std::unique_ptr <beast::asio::ssl_bundle>&& bundle,
-            http_request_type&& request,
-                boost::asio::ip::tcp::endpoint remote_address);
-
-    Handoff
-    onHandoff (Session& session, boost::asio::ip::tcp::socket&& socket,
         http_request_type&& request,
-            boost::asio::ip::tcp::endpoint remote_address);
+        boost::asio::ip::tcp::endpoint const& remote_address);
+
```

Comment:
Merging the two functions into one does resolve the code duplication. However, the `bundle` argument is now optional, and that isn't clear from the interface. I'd change that type to a `std::optional<std::unique_ptr<...>>`. I'd also pass the parameter as `std::none` instead of `{}` when it's missing.

Reply:
I thought about that but optional rvalue references are illegal.

Reply:
@seelabs Pushed an overload solution, let me know what you think.

Reply:
@miguelportilla You don't need an optional rvalue reference. Pass the `std::optional` by value and the optional should contain the `std::unique_ptr` by value.

Reply:
FWIW, here's a commit with the `boost::optional` interface: https://github.com/seelabs/rippled/commit/8e380837ca652a86dc6de64286dba20e731edfeb

Reply:
Actually, I have to think carefully about that change. In the old code, the `bundle` is actually conditionally moved from, in the new code it'll always be moved from. If the old code depends on the `bundle` still being valid after the call to `onHandle` then my suggestion is broken.

Reply:
Let's go with the overload solution, it's too risky to make the `boost::optinal` change because of the conditional move.